google-cloud-bigquery/lib/google/cloud/bigquery/job.rb

```diff
@@ -239,12 +244,16 @@ def reload!
         #                               format: "json"
         #   extract_job.wait_until_done!
         #   extract_job.done? #=> true
-        def wait_until_done!
-          backoff = ->(retries) { sleep 2 * retries + 5 }
+        def wait_until_done! min: 5, max: Float::INFINITY
```

Comment:
We have two "max" values that we've discusses to far. The one I'd like to see here is the max delay value. The maximum time between refresh calls.

The other "max" value we've discussed is the total amount of time until an error is wait until the job is completed. But we don't have this currently and haven't discussed the behavior of this value. In the current implementation you are just returning and therefore the job is not yet complete. So users will have to check the job status. I think this is wrong. If we have this value then it should raise an error, and raise it exactly when the timeout is reached. I would skip this max timeout value for now, since implementing it will likely take some additional effort.